### PR TITLE
Shrink unsafe blocks 

### PR DESCRIPTION
**Pull request by peamaeq**
The scope of the unsafe blocks can be appropriately reduced

### CODE REVIEW

File: src/condvar.rs [hunk 1]
```diff
@@ -197,11 +195,10 @@ impl Condvar {
 
     #[cold]
     fn notify_all_slow(&self, mutex: *mut RawMutex) -> usize {
-        unsafe {
             // Unpark one thread and requeue the rest onto the mutex
             let from = self as *const _ as usize;
             let to = mutex as usize;
-            let validate = || {
+            let validate = unsafe { || {
```
Comment (Amanieu) [inline src/condvar.rs:201]: Can this `unsafe` be moved inside the closure?

File: src/condvar.rs [hunk 2]
```diff
@@ -317,10 +312,10 @@ impl Condvar {
                     }
                     true
                 };
-                let before_sleep = || {
+                let before_sleep = unsafe { || {
```
Comment (Amanieu) [inline src/condvar.rs:315]: Same here.

File: src/condvar.rs [hunk 3]
```diff
@@ -136,11 +136,10 @@ impl Condvar {
...
     fn notify_one_slow(&self, mutex: *mut RawMutex) -> bool {
-        unsafe {
-            // Unpark one thread and requeue the rest onto the mutex
-            let from = self as *const _ as usize;
-            let to = mutex as usize;
-            let validate = || {
+        // Unpark one thread and requeue the rest onto the mutex
+        let from = self as *const _ as usize;
+        let to = mutex as usize;
+            let validate = unsafe { || {
```
Comment (Amanieu) [inline src/condvar.rs:142]: And here.

File: src/condvar.rs (hunk 1 above)
Comment (peamaeq) [reply src/condvar.rs]: Sorry for the long reply, very good suggestion

**Pull request merged**